### PR TITLE
Pull Request for Code Review 

### PR DESCRIPTION
**Pull request by kobiyoyo**

### CODE REVIEW

**Comment by JamesOkunlade** [comment]
## TSE working on it.

**Review by JamesOkunlade — CHANGES_REQUESTED** [review]
Hello @kobiyoyo ,

Good job on writing those enumerable methods. Some passed well but a few did not pass the test.

- `my_any?`, `my_none` and `my_inject` is not passing the test suite. For `my_any?` and `my_none?` you were expected to return a boolean but you are returning an array.

- my_any? should  return true if the block ever returns a value other than false or nil 
- my_none? should return true if the block never returns true for all elements 
- my_inject should combine all elements of enum by applying a binary operation, specified by a block 

Please work on those and also update your README.md file to contain the list of the methods you have written.

Request for another code review when you're done.

Happy coding!

**Comment by bellom** [comment]
## Currently reviewing this PR by Sean

**Review by bellom — COMMENTED** [review]
Hi @kobiyoyo, 
Good job using Git Flow.

Good work, according to the previous reviewer, you are to correct 3 methods, but two are working fine, remain `my_inject` method, throwing two argument error when running case test on it.
```     
ArgumentError:
 wrong number of arguments (given 0, expected 1)
```
You are to make use of one argument.
Kindly work on this and submit another code review....
Message on slack if you need help.

File: main.rb [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+            }
+        return new_arry
+    end
+
+    def my_inject(accum,&block)
+        self.my_each{|x|
+            accum = block.call(accum,x)
+            }
+        return accum
+    end
```
Comment (maelfosso) [inline main.rb:94]: Not working

```
>> [5, 6, 7, 8, 9, 10].my_inject { |product, n| product + n }
Traceback (most recent call last):
        7: from /home/maelfosso/.rvm/gems/ruby-2.6.0/bin/ruby_executable_hooks:24:in `<main>'
        6: from /home/maelfosso/.rvm/gems/ruby-2.6.0/bin/ruby_executable_hooks:24:in `eval'
        5: from /usr/share/rvm/rubies/ruby-2.6.0/bin/irb:23:in `<main>'
        4: from /usr/share/rvm/rubies/ruby-2.6.0/bin/irb:23:in `load'
        3: from /usr/share/rvm/rubies/ruby-2.6.0/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        2: from (irb):7
        1: from main.rb:71:in `my_inject'
ArgumentError (wrong number of arguments (given 0, expected 1))
>> [5, 6, 7, 8, 9, 10].inject { |product, n| product + n }
=> 45
>> 
```

File: main.rb [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        counter = 0
+        self.my_each do |x|
+            if !block_given?
+                return self.length
+            elsif yield(x) 
+                counter += 1
+            end
+        end
+        return counter
+    end
```
Comment (maelfosso) [inline main.rb:72]: Not working

```
>> ary = [1, 2, 4, 2]
=> [1, 2, 4, 2]
>> ary.count(2)
=> 2
>> ary.my_count(2)
Traceback (most recent call last):
        7: from /home/maelfosso/.rvm/gems/ruby-2.6.0/bin/ruby_executable_hooks:24:in `<main>'
        6: from /home/maelfosso/.rvm/gems/ruby-2.6.0/bin/ruby_executable_hooks:24:in `eval'
        5: from /usr/share/rvm/rubies/ruby-2.6.0/bin/irb:23:in `<main>'
        4: from /usr/share/rvm/rubies/ruby-2.6.0/bin/irb:23:in `load'
        3: from /usr/share/rvm/rubies/ruby-2.6.0/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        2: from (irb):17
        1: from main.rb:51:in `my_count'
ArgumentError (wrong number of arguments (given 1, expected 0))
```

File: main.rb [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    
+    def my_none?
+        self.my_each{|x|
+            if yield(x)
+                return false
+            else
+                return true
+            end  
+            }    
+    end
```
Comment (maelfosso) [inline main.rb:50]: Not working
```
>> %w{ant bear cat}.none? { |word| word.length >= 4 }
=> false
>> %w{ant bear cat}.my_none? { |word| word.length >= 4 }
=> true
>> 
```

File: main.rb [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    
+    def my_any?
+        self.my_each{|x|
+            if yield(x)
+                return true 
+            else
+                return false
+            end
+            }
+    end
```
Comment (maelfosso) [inline main.rb:40]: Not working
```
>> %w[ant bear cat].any? { |word| word.length >= 4 }
=> true
>> %w[ant bear cat].my_any? { |word| word.length >= 4 }
=> false
>> 
```

File: main.rb [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+        }
+        return new_array
+    end
+    
+    def my_all?
+        self.my_each{|x|
+            return false if !yield(x)
+            return true
+            }
+    end
```
Comment (maelfosso) [inline main.rb:30]: Not working 
```
>> [nil, true, 99].all?
=> false
>> [nil, true, 99].my_all?
Traceback (most recent call last):
       11: from /home/maelfosso/.rvm/gems/ruby-2.6.0/bin/ruby_executable_hooks:24:in `<main>'
       10: from /home/maelfosso/.rvm/gems/ruby-2.6.0/bin/ruby_executable_hooks:24:in `eval'
        9: from /usr/share/rvm/rubies/ruby-2.6.0/bin/irb:23:in `<main>'
        8: from /usr/share/rvm/rubies/ruby-2.6.0/bin/irb:23:in `load'
        7: from /usr/share/rvm/rubies/ruby-2.6.0/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        6: from (irb):33
        5: from main.rb:25:in `my_all?'
        4: from main.rb:3:in `my_each'
        3: from main.rb:3:in `each'
        2: from main.rb:4:in `block in my_each'
        1: from main.rb:26:in `block in my_all?'
LocalJumpError (no block given (yield))
>> 
>> [].all?
=> true
>> [].my_all?
=> 0...0
>> 
```

**Comment by maelfosso** [comment]
Hello,

You did a really nice job.

- [x] Please, indent your code

- [x] There are some enumerable functions which work when no block is given. Could you try to take them into consideration, please?

- [x] Check the result from some test put above and correct the corresponding functions.

Kindly make those changes and submit a new code review request.

**Review by Oghenebrume50 — APPROVED** [review]
Great work on this, everything works now. please merge and submit the completion form

**Pull request merged**